src/execnet/gateway_io.py

```diff
@@ -53,8 +56,22 @@ def shell_split_path(path):
     return shlex.split(path)
 
 
+def is_frozen_environment():
+    """Check if running in a Frozen environment."""
+    return getattr(sys, "frozen", False) and hasattr(sys, "_MEIPASS")
+
+
+def get_python_executable():
+    """Get the correct Python interpreter path."""
+    if is_frozen_environment():
+        return os.environ.get("PYTHON_EXECUTABLE", "python")
```

Comment:
As I mentioned, not clear to me why this is needed, seems to me we should always return `sys.executable` from here?